bundles/binding/org.openhab.binding.velux/src/main/java/org/openhab/binding/velux/bridge/VeluxBridgeDeviceStatus.java

```diff
@@ -19,35 +20,85 @@
+     * <li>{@link #gwStateDescription} containing the verbose gateway state.
+     * </ul>
+     */
+    public class Channel {
+        public boolean isRetrieved = false;
+        public StringType gwState;   
+        public StringType gwStateDescription;   
+    }
+
+    private Channel channel = null;
+
+	// Constructor methods
+    
+	/**
+	 * Constructor.<P>
+	 * Initializes the internal data structure {@link #channel} of Velux actuators/products,
+	 * which is publicly accessible via the method {@link #getChannel()}.
+	 */
+	public VeluxBridgeDeviceStatus() {
+		logger.trace("VeluxBridgeDeviceStatus(constructor) called.");
+		channel = new Channel();
+	}
+
+	// Class access methods
+	
+	/**
+	 * Provide access to the internal structure of the device status.
+	 * @return <b>channel</b>
+	 *         of type {@link Channel} describing the overall actual device status.
+	 */
+	public Channel getChannel() {
+		return channel;
+	}
+
 
     /**
-     * Login into bridge, query the bridge for device status and logout from bridge
-     * based on a well-prepared environment of a {@link VeluxBridgeProvider}.
+     * Complete workflow for retrieving the firmware version, consisting of Login into bridge, querying the firmware
+     * version and logout from bridge based on a well-prepared environment of a {@link VeluxBridgeProvider}, where the
+     * results are stored within as well in {@link VeluxBridgeDeviceStatus#channel}.
      *
      * @param bridge Initialized Velux bridge handler.
-     * @return <b>status</b> of type String describing the status of the device/bridge.
+     * @return <b>channel</b> - or null -
+     *         of type {@link VeluxBridgeDeviceStatus.Channel} describing the overall result of this interaction.
      */
+	public Channel retrieve(VeluxBridge bridge) {
+		logger.trace("retrieve() called.");
 
-    public String retrieve(VeluxBridgeProvider bridge) {
-        logger.trace("retrieve() called.");
-
-        logger.trace("retrieve() About to query device status.");
-        BCgetDeviceStatus.Response response = bridge.bridgeCommunicate(new BCgetDeviceStatus());
-        if ((response == null) || (!response.getResult())) {
-            logger.trace("retrieve() finished with failure.");
-            return null;
-        }
-        logger.trace("retrieve() finished successfully with result {}.", response.getDeviceStatus());
-        return response.getDeviceStatus();
-    }
+		logger.trace("retrieve() About to query device status.");
+		GetDeviceStatus bcp = bridge.bridgeAPI().getDeviceStatus();
+		if ((bridge.bridgeCommunicate(bcp)) && (bcp.isCommunicationSuccessful())) {
+			VeluxGwState state = bcp.getState();
+			this.channel.gwState = new StringType(state.toString());
+			this.channel.gwStateDescription = new StringType(state.toDescription());
+			this.channel.isRetrieved = true;
+			logger.trace("retrieve() finished successfully with result {}.", state.toDescription());
+			return channel;
+		}
+		else {
```

Comment:
I don't think so. If the method had been called with success recently, it would return the old obsoleted information if it would return channel.
If the methods returns null, the bridge query had failed and no valid device status is existing.